linera-execution/src/lib.rs

```diff
@@ -312,6 +315,8 @@ pub struct CalleeContext {
 pub struct QueryContext {
     /// The current chain id.
     pub chain_id: ChainId,
+    /// The height of the next block on this chain.
+    pub next_block_height: BlockHeight,
```

Comment:
Could this be the current block height?

Reply:
Then we'd need to make it an `Option`… and we use the next block height in the chain state itself, so to me it feels more natural.

It currently isn't used at all for queries, but of course we might use it somehow in the future. Queries would then be executed as if they were part of the _next_ block, rather than appended after the previous block. I'm happy to change it, if you think that's unintuitive.

Reply:
I'm not sure I understand why it would need to be an `Option`. If the problem is representing an empty chain, I'm wondering if an empty chain can be queried at all?

Reply:
It does have a state, and maybe we'll at some point even register applications immediately on `OpenChain`, without a block. So I think it can be queried, yes.

Reply:
I see, thanks! In that case I think `Option<BlockHeight>` is more intuitive, but we can leave it as is and change again later :man_shrugging:

Reply:
We've been pretty consistent on this so far